### DISCUSSION TITLE
What's the correct way to put a gradient on line?

### DISCUSSION THREAD

**Discussion by Max-H-0** [AvaloniaUI/Avalonia#14580] [Q&A]
I'm trying to put a gradient on a line that goes from its start point to the end point but I can't figure out how to do it.

My first idea was to simply use the same points
```xaml                
                <Line Name="PART_Visual"
                          StartPoint="{Binding Output.Position}"
                          EndPoint="{Binding Input.Position}"
                          StrokeThickness="20"
                          StrokeJoin="Round"
                          Opacity="0.5">
                        <Line.Stroke>
                            <LinearGradientBrush StartPoint="{Binding Output.Position}"
                                                                EndPoint="{Binding Input.Position}">
                                <LinearGradientBrush.GradientStops>
                                    <GradientStop Offset="0" Color="Red"/>
                                    <GradientStop Offset="1" Color="Green"/>
                                </LinearGradientBrush.GradientStops>
                            </LinearGradientBrush>
                        </Line.Stroke>
                </Line>
```

but that didnt quite work:
https://github.com/AvaloniaUI/Avalonia/assets/81490420/e57cd9db-09eb-43f4-9056-5442adaca016

My second idea was to use a radial gradient and set the center to the start point and the radius using a converter based on the distance between the points but I couldn't get that to work either.

What's the correct way to do this?

**Comment by Gillibald** [comment]
Change start and end point of the brush like so `<LinearGradientBrush StartPoint="0%,0%" EndPoint="100%,100%">`

**Reply by Max-H-0** [reply]
Thank you but that's sadly not looking quite right either
https://github.com/AvaloniaUI/Avalonia/assets/81490420/ecdb3593-94d9-4e9a-a9bf-74fae3841cfa

**Comment by kekekeks** [comment]
We currently have a bug that's been around for a _decade_ with RelativeUnit.Relative handling  for graphics primitives that don't originate at 0:0, e. g. everything drawn at non-0:0 origin using DrawingContext _and_ `Line` Shape since it actually uses StartPoint for its DrawingContext call.

Good news is that it's currently being worked on.

**Reply by Max-H-0** [reply]
Great to hear it's being worked on.
Do you know if there are any workarounds for now or when it's roughly going to be fixed?
